### PR TITLE
Allow Bazel 6 users to explicitly opt-in to rules_apple 2.x

### PR DESCRIPTION
**Pull request by thiagohmcruz**
Regressed in https://github.com/bazel-ios/rules_ios/pull/773

### CODE REVIEW

**Review by luispadron — APPROVED** [review]

File: rules/repositories.bzl [hunk 1]
```diff
@@ -149,7 +151,7 @@ def _rules_ios_bzlmod_dependencies():
             ],
             sha256 = "c6966ec828da198c5d9adbaa94c05e3a1c7f21bd012a0b29ba8ddbccb2c93b0d",
         )
-    elif bazel_version.major == "6" and bazel_version.minor == "1":
```
Comment (justinseanmartin) [inline rules/repositories.bzl:152]: @thiagohmcruz and I discussed offline - This feels like a weird special behavior to have for handling the CI use-case ([per this comment](https://github.com/bazel-ios/rules_ios/pull/773/files#r1362497302)). I like that the proposed change makes the rules_apple 2.x support a bit more explicit in the API contract. Before this change, someone using 6.1 and accidentally getting some weird default dependencies would probably be confusing.

I think we're going to omit the Bazel version check here and have this be defaults that we know work for rules_apple 2.x based on an explicit `use_rules_apple_2 = True` param. We're also going to reset this back to the values used prior to #773 (rules_apple v2.5.0 and rules_swift v1.10.0) as the defaults. For CI, Thiago is going to mutate the WORKSPACE file in the CI script to pass `use_rules_apple_2 = True`. Separately to this, we could bump everything to run on Bazel 6.4 as well and things should continue to work.

**Comment by thiagohmcruz** [comment]
@justinseanmartin To confirm the `sed` to use `rules_apple` 2.x on CI + Bazel 6.1.2 I checked the events [here](https://app.buildbuddy.io/invocation/bad5e9e7-c8c2-4023-9652-de9902368b49#raw) and saw this:
```json
  {
    "id": {
      "fetch": {
        "url": "https://github.com/bazelbuild/rules_apple/releases/download/2.5.0/rules_apple.2.5.0.tar.gz"
      }
    },
    "fetch": {
      "success": true
    }
  },
```

**Review by justinseanmartin — APPROVED** [review]
It might be worth adding a comment on on the elsif block and updating the one on the else block. You technically don't need to specify `load_rules_apple_2_dependencies` in order to load the rules_apple 2.x, given the `_maybe()`'s, this is just a convenience for folks that want to use the defaults and making the rules_apple 2.x support a bit more clear in the API contract.

File: rules/repositories.bzl (hunk 1 above)
Comment (luispadron) [reply rules/repositories.bzl:152]: Thank you both, that's easier to reason about

**Review by mattrobmattrob — APPROVED** [review]
Think this should be a major version bump since we're changing the public API? I'm curious how many folks are using this definition vs. the `maybe` support outside of Block.

**Comment by luispadron** [comment]
> Think this should be a major version bump since we're changing the public API? I'm curious how many folks are using this definition vs. the `maybe` support outside of Block.

I personally wouldn't consider that public api, IMO it shouldn't have been merged to begin with since again users in Bazel can simply define the version they want to use. Ideally, we wouldn't have this at flag all. 

If we wanted to test in CI with 6.x and rules_apple 2 we can configure that in CI

**Comment by thiagohmcruz** [comment]
+1 to treating this as an improved contract but not necessarily a major version bump (no strong opinions though if folks want to do so). To hard code it to "Bazel 6.1.x" was a bit hack-y and would bite folks on that version without them knowing. With this new flag at least it's now an opt-in for "I'm using whatever Bazel version <=6.x.x and I want to use `rules_apple` 2.x".

**Comment by mattrobmattrob** [comment]
I don't feel strongly. Please just call it out in the release notes. 🙏

**Pull request merged**